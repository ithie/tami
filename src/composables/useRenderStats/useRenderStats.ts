import { computed, type Ref } from 'vue'

const max = 10

const renderProgressPart = (element: string, amount: number): string =>
  new Array(amount).join(element)

const renderProgress = (amount: number) =>
  `${amount <= 0 ? '' : renderProgressPart('â–ˆ', amount + 1)}${amount >= max ? '' : renderProgressPart(' ', max - amount + 1)}`

export default ({
  hunger,
  happieness,
  energy
}: {
  hunger: Ref<number>
  happieness: Ref<number>
  energy: Ref<number>
}) => {
  return {
    renderedHunger: computed(() => renderProgress(hunger.value)),
    renderedHappieness: computed(() => renderProgress(happieness.value)),
    renderedEnergy: computed(() => renderProgress(energy.value))
  }
}
