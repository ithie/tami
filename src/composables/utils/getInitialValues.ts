import { ref, type Ref } from 'vue'

export default <T = Ref<number>>(
  [energy, happieness, hunger]: [number, number, number],
  noRef = false
): { energy: T; happieness: T; hunger: T } => ({
  energy: (noRef ? energy : ref(energy)) as T,
  happieness: (noRef ? happieness : ref(happieness)) as T,
  hunger: (noRef ? hunger : ref(hunger)) as T
})
