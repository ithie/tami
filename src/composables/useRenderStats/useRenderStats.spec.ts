import { describe, expect, test } from 'vitest'
import useRenderStats from './useRenderStats'
import { type Ref } from 'vue'
import getInitialValues from '../utils/getInitialValues'

const assertions: { hunger: Ref<number>; happieness: Ref<number>; energy: Ref<number> }[] = [
  getInitialValues([0, 5, 5]),
  getInitialValues([10, 15, 15]),
  getInitialValues([-10, 3, 3])
]

describe('composables/useRenderStats', () => {
  assertions.forEach((given, index) =>
    test(`should render stats: ${index}`, () => {
      const { renderedEnergy, renderedHappieness, renderedHunger } = useRenderStats(given)
      expect(renderedEnergy.value).toMatchSnapshot()
      expect(renderedHappieness.value).toMatchSnapshot()
      expect(renderedHunger.value).toMatchSnapshot()
    })
  )
})
