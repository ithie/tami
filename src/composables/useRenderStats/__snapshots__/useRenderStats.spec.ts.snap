// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`composables/useRenderStats > should render stats: 0 1`] = `"          "`;

exports[`composables/useRenderStats > should render stats: 0 2`] = `"█████     "`;

exports[`composables/useRenderStats > should render stats: 0 3`] = `"█████     "`;

exports[`composables/useRenderStats > should render stats: 1 1`] = `"██████████"`;

exports[`composables/useRenderStats > should render stats: 1 2`] = `"███████████████"`;

exports[`composables/useRenderStats > should render stats: 1 3`] = `"███████████████"`;

exports[`composables/useRenderStats > should render stats: 2 1`] = `"                    "`;

exports[`composables/useRenderStats > should render stats: 2 2`] = `"███       "`;

exports[`composables/useRenderStats > should render stats: 2 3`] = `"███       "`;
